chainer/link.py

```diff
@@ -376,6 +378,20 @@ def to_gpu(self, device=None):
         self._cpu = False
         return self
 
+    def to_intel64(self):
```

Comment:
Do we have to provide `to_intel64` method, even though the use of iDeep is not determined by array data type?
I feel it is better for users not to be aware of difference of array types, as ideep.mdarray proxies methods to numpy array (and it now supports operator overload).
(The only concern is that ideep.mdarray does not support `__str__`/`__repr__` so that it shows `<Swig Object of type 'std::vector< mdarray >::value_type *' at 0x7f2011708650>` instead of np.array2string output when used interactively. Maybe we can ask Intel guys to implement this...)

Reply:
It is needed in optimzers (See #4285).
When updating a parameter, iDeep will be used only when its underlying array is `mdarray`.
It's not automatically converted.

Reply:
Are there any problem if `should_use_ideep` is used in `update_core_cpu` of Optimizers?
(Maybe I am missing something...)

Reply:
The point is `update_core_cpu` updates the array *in-place*. Thus the array has to be converted to `mdarray` prior to updates.

Reply:
Yes, I think that's correct.